src/floyd-warshall.jl

```diff
@@ -36,17 +36,17 @@ end
 #     Note that it is possible to consume large amounts of memory as the
 #     space required for the FloydWarshallState is O(n^2).
 #     """ ->
-function floyd_warshall_shortest_paths(
+function floyd_warshall_shortest_paths{T<:Real}(
     g::AbstractGraph;
-    edge_dists::AbstractArray{Float64, 2} = Array(Float64,(0,0))
+    edge_dists::AbstractArray{T,2} = zeros(0,0)
 )
 
     use_dists = issparse(edge_dists)? nnz(edge_dists > 0) : !isempty(edge_dists)
     n_v = nv(g)
-    dists = fill(convert(Float64,Inf), (n_v,n_v))
-    parents = zeros(Int, (n_v,n_v))
+    S = typeof(one(T) + one(T))
+    dists = fill(typemax(S), n_v, n_v)
```

Comment:
The issue is that this will all break horribly if someone decides to use UInt8, and we have no "max test" like isinf for these other types (except for constructing something with `typemax` and kludging around that way).

I've avoided type parameterization in LightGraphs precisely because of the complexity it introduced with Graphs.jl. LightGraphs is designed to be quick, easy to understand, but not as flexible as Graphs.